src/qutip_tensorflow/core/data/add.py

```diff
@@ -43,18 +43,20 @@ def iadd_tftensor(left, right, scale=1):
 
 def sub_tftensor(left, right):
     _check_shape(left, right)
-    return TfTensor._fast_constructor(left._tf - right._tf, shape=left.shape)
+    return left._fast_constructor(left._tf - right._tf, shape=left.shape)
 
 
 # `add_conversions` will register the data layer
 qutip.data.add.add_specialisations(
     [
-        (TfTensor, TfTensor, TfTensor, add_tftensor),
+        (TfTensor128, TfTensor128, TfTensor128, add_tftensor),
```

Comment:
Suggestion: If we write a function that adds the specializations, we can avoid duplicating everything and we can make it easier to register the specializations separately if for some reason someone wants one and not the other.

E.g.
```python
def _add_specialisations(cls):
    qutip.data.add.add_specialisations([
        (cls, cls, cls, add_tftensor),
    ])

_add_specialisations(TfTensor128)
_add_specialisations(TfTensor64)
```

Reply:
One downside is that the specialisations are currently added a bit all over the place (which is not necessarily a terrible thing), so the simple approach above will need a bit more thought.

Reply:
I think I will stick with the simple approach here just because I only need to add one more specialisation each time. Although it may be necessary to consider such function if we want to bring support for multiple backends to qutip-tensornetwork.